ast/src/parsed/mod.rs

```diff
@@ -306,6 +317,52 @@ impl<R> Children<Expression<R>> for EnumVariant<Expression<R>> {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct TraitDeclaration<E = u64> {
+    pub name: String,
+    pub type_vars: TypeBounds,
```

Comment:
I'm wondering, I think we will really just allow a list of type variables for a trait declaration, so maybe just a vector of strings?

Reply:
If it's easier to use TypeBounds then it's also fine!